scripts/buildWebtypes.js

```diff
@@ -0,0 +1,281 @@
+const additionalAttributes = [
+  // Theme attribute is not properly declared at the moment
+  {
+    name: 'theme',
+    description: 'The theme variants to apply to the component.',
+    type: 'string | null | undefined',
+  },
+];
+
+/**
+ * Get packages using lerna, excluding blacklisted packages
+ */
+function getRelevantPackages() {
+  const output = execSync('./node_modules/.bin/lerna ls --json --loglevel silent');
+  const allPackages = JSON.parse(output.toString()).map((project) => project.name.replace('@vaadin/', ''));
+  return allPackages.filter((pkg) => !blacklistedPackages.some((blacklistedPackage) => pkg.match(blacklistedPackage)));
+}
+
+function loadAnalysis() {
+  const analysisPath = path.resolve('./analysis.json');
+  try {
+    return JSON.parse(fs.readFileSync(analysisPath, 'utf8'));
+  } catch (e) {
+    throw new Error(
+      `Could not read output of the Polymer Analyzer from: ${analysisPath}. Make sure to run the Polymer Analyzer before generating web-types.`,
+    );
+  }
+}
+
+/**
+ * Maps a type string such as `'boolean | null | undefined'` into an array of type
+ * strings, such as `['boolean', 'null', 'undefined']`
+ * @param typeString
+ * @returns {string[]}
+ */
+function mapType(typeString) {
+  const sanitizedTypeString = (typeString || '').replace(/[!()]/g, '');
+  const types = sanitizedTypeString.split('|');
+  return types.map((type) => type.trim());
+}
+
+/**
+ * Transforms a description text of an element, attribute, property, event to:
+ * - make relative docs links point to the absolute documentation URL
+ * @param packageJson
+ * @param description
+ * @returns {*}
+ */
+function transformDescription(packageJson, description) {
+  // Transform relative documentation links to absolute ones
+  description = description.replace(
+    /\(#\/elements\/(.*)\)/g, // Matches "(#/elements/$1)"
+    `(${API_DOCS_BASE_PATH}/${packageJson.version}/#/elements/$1)`,
+  );
+  // Remove multiple newlines between subsequent code examples,
+  // The IntelliJ markdown renderer will otherwise collapse
+  // both examples into one
+  description = description.replace(/```(\n)+```/g, '```\n```');
+
+  return description;
+}
+
+function camelize(text) {
+  return text.replace(/-./g, (x) => x[1].toUpperCase());
+}
+
+function isWritablePrimitiveAttribute(elementAnalysis, attribute) {
+  // Attributes do not have metadata, so we need to look at the corresponding
+  // property
+  const propertyName = camelize(attribute.name);
+  const matchingProperty = elementAnalysis.properties.find((property) => property.name === propertyName);
+  // If we can not find the property, just include the attribute rather than exclude
+  if (!matchingProperty) {
+    return true;
+  }
+  const isWritable = !matchingProperty.metadata.polymer.readOnly;
+  const hasPrimitiveType =
+    !matchingProperty.metadata.polymer.attributeType ||
+    ['String', 'Number', 'Boolean'].includes(matchingProperty.metadata.polymer.attributeType);
```

Comment:
> It would be nice to filter out all the object properties, to not show them as attributes. Example:

Added this filter that should exclude all attributes where the matching property is not have a primitive attribute type.